src/device.cc

```diff
@@ -43,6 +43,15 @@ namespace hpp {
 namespace manipulation {
 using ::pinocchio::Frame;
 
+DevicePtr_t Device::create(const std::string& name) {
+  Device* ptr = new Device(name);
+  DevicePtr_t shPtr(ptr);
+  ptr->init(shPtr);
+  return shPtr;
```

Comment:
Why not `make_shared` ??

Reply:
As the constructor is protected, `std::make_shared` cannot be used in a straightforward way.
There are some obscure workarounds though: https://stackoverflow.com/questions/8147027/how-do-i-call-stdmake-shared-on-a-class-with-only-protected-or-private-const/8147213#8147213